File: java/Samples/Biometrics/VB/AbisSampleVB/Settings/SettingsPageBase.vb
Imports Microsoft.VisualBasic
Imports Neurotec.Biometrics.Client
Imports System

Partial Public Class SettingsPageBase
	Inherits Neurotec.Samples.PageBase
	#Region "Public methods"

	Public Sub New()
		InitializeComponent()
	End Sub

	#End Region

	Private privateClient As NBiometricClient
	Protected Property Client() As NBiometricClient
		Get
			Return privateClient
		End Get
		Set(ByVal value As NBiometricClient)
			privateClient = value
		End Set
	End Property

	#Region "Public methods"

	Public Overrides Sub OnNavigatedTo(ParamArray ByVal args() As Object)
		If args Is Nothing OrElse args.Length <> 1 Then
			Throw New ArgumentException("args")
		End If
		Client = CType(args(0), NBiometricClient)
		If Client Is Nothing Then
			Throw New ArgumentException()
		End If

		LoadSettings()
		MyBase.OnNavigatedTo(args)
	End Sub

	Public Overrides Sub OnNavigatingFrom()
		SaveSettings()
		Client = Nothing

		MyBase.OnNavigatingFrom()
	End Sub

	Public Overridable Sub LoadSettings()
	End Sub

	Public Overridable Sub SaveSettings()
	End Sub

	Public Overridable Sub DefaultSettings()
		LoadSettings()
	End Sub

	#End Region
End Class

File: java/Samples/Biometrics/CS/AbisSampleCS/Settings/SettingsPageBase.cs
using System;
using Neurotec.Biometrics.Client;

namespace Neurotec.Samples
{
	public partial class SettingsPageBase : Neurotec.Samples.PageBase
	{
		#region Public methods

		public SettingsPageBase()
		{
			InitializeComponent();
		}

		#endregion

		#region Public properties

		protected NBiometricClient Client { get; set; }

		#endregion

		#region Public methods

		public override void OnNavigatedTo(params object[] args)
		{
			if (args == null || args.Length != 1) throw new ArgumentException("args");
			Client = (NBiometricClient)args[0];
			if (Client == null) throw new ArgumentException();

			LoadSettings();
			base.OnNavigatedTo(args);
		}

		public override void OnNavigatingFrom()
		{
			SaveSettings();
			Client = null;

			base.OnNavigatingFrom();
		}

		public virtual void LoadSettings()
		{
		}

		public virtual void SaveSettings()
		{
		}

		public virtual void DefaultSettings()
		{
			LoadSettings();
		}

		#endregion
	}
}
